### PR TITLE
feat: CI integrations Logger improvements

### LINKED ISSUE #995: Metrics CLI: investigate logger design improvement to solve "prefix methods"
**Description**
We want to improve Metrics CLI Logger design to init instance with the prefix (using runtimeType?) so that we don't have to implement such methods for every class:

image

**Acceptance Criteria**
- [ ] Design updated with an approach that solves "prefix methods" or arguments presented against it

cc: @solid-vovabeloded

### PR DESCRIPTION
**Pull request by solid-vladlaba**
Fixes #995

## Checklist

- [x] Documented
- [x] Unit tested
- [x] Component tested

### CODE REVIEW

File: metrics/ci_integrations/test/cli/logger/logger_test.dart [hunk 1]
```diff
@@ -1,65 +1,197 @@
...
+          writer: writerMock,
+          verbose: testVerbose,
+        ),
+        returnsNormally,
+      );
     });
 
     test(
-      ".logMessage() logs the given message to the message sink",
+      ".forClass() returns a new logger if LoggerManager does not contain any for the given class type",
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/cli/logger/logger_test.dart:94]: Maybe we can just test that this method delegates to the `LoggerManager.getLogger()`? It seems like we've already tested that the `getLogger()` works correctly in `LoggerManager` tests.

File: metrics/ci_integrations/test/cli/logger/logger_test.dart [hunk 2]
```diff
@@ -1,65 +1,197 @@
...
+          (log) => matchesPrefixes(log, '$testSourceClass', message),
+        );
+
+        verify(writerMock.write(argThat(logMatcher))).called(1);
       },
     );
   });
 }
+
+class _ClassStub {}
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/cli/logger/logger_test.dart:197]: It seems like this class lacks documentation. Please, go through all tests and apply this suggestion.

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-vladlaba! I've added a couple of suggestions here - please take a look. Overall, the new logger looks great - thanks for improving it!

File: metrics/ci_integrations/test/cli/logger/logger_test.dart (hunk 1 above)
Comment (solid-vladlaba) [reply metrics/ci_integrations/test/cli/logger/logger_test.dart:94]: Resolve this as we discussed via chat

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-vladlaba!

**Pull request merged**